### PR TITLE
feat: add filter component

### PR DESCRIPTION
**Pull request by chrishiguto**
### About

This PR adds a Filter component that receives a `filters` prop to render fields that are supposed to filter given data. 
The filter object provided to filters prop accepts three types of fields:

```
1. FilterType.Text,
2. FilterType.Autocomplete,
3. FilterType.Select
```

Each of these types have different required props. Please check the types to know which props are required for each type of field.

This PR also adds:

 - AutocompleteField
 - SelectField

### What is not in scope

It's not in the scope of this PR to integrate this with table/data provider because we still need to work on useTable hook to be able to use it inside of other components. Currently every time useTable gets invoked it gets new data created because it's not tied to any React Context.

### Usage example

```
<Filter
        filters={[
          {
            type: FilterType.Text,
            defaultValue: 'default value',
            onChange: 'on change function that will trigger filter handler',
          },
          {
            type: FilterType.Select,
            label: 'Select label',
            options: 'array of SelectOption',
            isLoading: 'boolean value',
            defaultValue: 'default value',
            onChange: 'on change function that will trigger filter handler',
          },
          {
            type: FilterType.Autocomplete,
            columns: 4,
            currentValue: 'current value of autocomplete',
            isLoading: 'boolean value',
            defaultValue: 'default value - selectOption',
            options: 'array of SelectOption',
            onChange: 'on change function that will trigger filter handler',
          },
        ]}
      />
```

### Column prop

Each object accepts a `column` prop that leverages MUI grid to render the field.

```
{
            type: FilterType.Text,
            defaultValue: 'default value',
            column: 2,
            onChange: 'on change function that will trigger filter handler',
          },
```

### CODE REVIEW

**Comment by chrishiguto** [comment]
@andreneto97 Unfortunately we still need to pass `onChange` to the filters object because we can't handle filter state/update inside this component because we can't use `useTable`.
We'd need to first work on `useTable` to share context before doing this.

File: packages/react-material-ui/src/components/AutocompleteField/AutocompleteField.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}: AutocompleteFieldProps) => {
+  const handleChange = (
+    _: SyntheticEvent<Element, Event>,
+    newValue: SelectOption | null,
+  ) => {
+    onChange(newValue?.value ?? null);
+  };
+
+  if (isLoading) {
+    return <Box width={300}>{/*  <FormFieldSkeleton hideLabel /> */}</Box>;
```
Comment (andreneto97) [inline packages/react-material-ui/src/components/AutocompleteField/AutocompleteField.tsx:34]: ??

File: packages/react-material-ui/src/components/SelectField/SelectField.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return `${label}: ${optionLabel}`;
+  };
+
+  const handleChange = (event: SelectChangeEvent<string>) => {
+    const value = event.target.value as string;
+    onChange(getStatusValue(value));
+  };
+
+  if (isLoading) {
+    return <Box width={190}>{/* <FormFieldSkeleton hideLabel /> */}</Box>;
```
Comment (andreneto97) [inline packages/react-material-ui/src/components/SelectField/SelectField.tsx:57]: ???

Comment (chrishiguto) [reply packages/react-material-ui/src/components/SelectField/SelectField.tsx]: FormFieldSkeleton depends on #36 
I've merged and fixed this

**Pull request merged**